src/Stack/Build/ConstructPlan.hs

```diff
@@ -473,22 +350,29 @@ addDep name = do
             updateLibMap name res
             return res
 
--- FIXME what's the purpose of this? Add a Haddock!
-tellExecutables :: PackageName -> PackageSource -> M ()
+-- | Update the location of executables (if any) in the pdInstall map.
+tellExecutables :: PackageName ->
+    -- | The package source for which the exe locations should (possibly) be updated.
```

Comment:
Very minor thing: I think normally we use postfix and not prefix haddocks for args i.e. `-- ^`
It's better to make code more consistent